tests/Manifest/ManifestManager/Options/OutTableManifestOptionsTest.php

```diff
@@ -0,0 +1,184 @@
+                            'key' => 'another.arbitrary.key',
+                            'value' => 'A different value',
+                        ],
+                    ],
+                    'column_metadata' => [
+                        'column1' => [
+                            [
+                                'key' => 'yet.another.key',
+                                'value' => 'Some other value',
+                            ],
+                        ],
+                    ],
+                ],
+                (new OutTableManifestOptions())
+                    ->setEnclosure('_')
+                    ->setDelimiter('|')
+                    ->setColumnMetadata([
+                        'column1' => [
+                            [
+                                'key' => 'yet.another.key',
+                                'value' => 'Some other value',
+                            ],
+                        ],
+                    ])
+                    ->setColumns(['id', 'number', 'other_column'])
+                    ->setDestination('my.table')
+                    ->setIncremental(true)
+                    ->setMetadata([
+                        [
+                            'key' => 'an.arbitrary.key',
+                            'value' => 'Some value',
+                        ],
+                        [
+                            'key' => 'another.arbitrary.key',
+                            'value' => 'A different value',
+                        ],
+                    ])
+                    ->setPrimaryKeyColumns(['id']),
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideInvalidOptions
+     */
+    public function testInvalidOptions(string $expectedExceptionMessage, callable $callWithInvalidArguments): void
+    {
+        $this->expectException(OptionsValidationException::class);
+        $this->expectExceptionMessage($expectedExceptionMessage);
+
+        $callWithInvalidArguments();
+    }
+
+    /**
+     * @return mixed[][]
+     */
+    public function provideInvalidOptions(): array
+    {
+        return [
+            [
+                'Each metadata item must be an array',
+                function (): void {
+                    (new OutTableManifestOptions())->setMetadata([
+                        'one',
+                        'two',
+                    ]);
+                },
+            ],
+            [
+                'Each metadata item must have only "key" and "value" keys',
+                function (): void {
+                    (new OutTableManifestOptions())->setMetadata([
+                        [
+                            'key' => 'my-key',
+                            'value' => 'my-value',
+                            'something' => 'my-value',
+                        ],
+                    ]);
+                },
+            ],
+            [
+                'Each metadata item must have only "key" and "value" keys',
```

Comment:
[
                'Each metadata item must have only "key" and "value" keys',
                function (): void {
                    (new OutTableManifestOptions())->setMetadata([
                        [
                            'value' => 'my-value',
                            'key' => 'my-key',
                        ],
                    ]);
                },
            ],